### PR TITLE
feat: slice by name when there are duplicates

### PR DESCRIPTION
**Pull request by keviny2**
Issue https://github.com/BiocPy/SummarizedExperiment/issues/21

### CODE REVIEW

File: src/summarizedexperiment/utils/_slicer.py [hunk 1]
```diff
@@ -9,34 +9,31 @@
...
-        raise TypeError(f"{_target} is not a collection") from exception
-
-    missing_names = target.difference(source)
-    if not missing_names.empty:
-        raise ValueError("invalid index name(s): " + ", ".join(missing_names))
-
-    return source.get_indexer(target).tolist()
+    missing_names = [value for value in target if value not in source]
+    if len(missing_names) > 0:
+        raise ValueError("invalid index names(s): " + ", ".join(missing_names))
```
Comment (jkanche) [inline src/summarizedexperiment/utils/_slicer.py:29]: using sets is much better than iteration

`missing_names = set(source).difference(target)`

Comment (keviny2) [reply src/summarizedexperiment/utils/_slicer.py]: this works
`missing_names = set(target).difference(source)`

File: src/summarizedexperiment/utils/_slicer.py [hunk 2]
```diff
@@ -9,34 +9,31 @@
...
+
+    value_to_index = {}
+    for index, value in enumerate(source):
+        if value not in value_to_index:
+            value_to_index[value] = index
+    
+    return [value_to_index[value] for value in target]
```
Comment (jkanche) [inline src/summarizedexperiment/utils/_slicer.py:36]: we should vectorize whenever possible. iterations in any language are slow.

**Comment by jkanche** [comment]
I realize something is wrong with your local repo. we don't have 20 commits to commit here. should just be the last two.

**Comment by keviny2** [comment]
> I realize something is wrong with your local repo. we don't have 20 commits to commit here. should just be the last two.

yeah i'm not sure why it did that...

**Comment by jkanche** [comment]
> > I realize something is wrong with your local repo. we don't have 20 commits to commit here. should just be the last two.
> 
> yeah i'm not sure why it did that...

I recommend resetting your repo, otherwise this is going to continue.

**Pull request closed without merging**